cpp/tests/community/louvain_test.cpp

```diff
@@ -29,7 +29,8 @@
 #include <vector>
 
 struct Louvain_Usecase {
-  bool test_weighted_{false};
```

Comment:
So, are we always testing for weighted graphs only?

Reply:
Yes.  Louvain requires a weighted graph.  All of the computations operate on the edge weights.

I'll add a `CUGRAPH_EXPECTS` in the algorithm implementation to check this, since we don't currently check that.

Reply:
Can't we implicitly assume edge weights of 1 if the input graph is unweighted? Or no real use case for this?

Reply:
I assume finding communities in unweighted graphs to be something practical.

Reply:
The python layer will add a weight of 1 for edges if the input graph is unweighted.

I think it would make the C++ code unwieldy to allow the input graph to be unweighted.  Once we coarsen the graph it must be weighted, so we'd end up having to have different code paths for the original graph than from the coarsened graphs.

Reply:
Gotcha, just wanted to point out that there is memory requirement vs implementation complexity trade-off.